lib/db/mem.js

```diff
@@ -284,6 +301,17 @@ module.exports = function (log, error) {
               deviceInfo.sessionTokenId = device.sessionTokenId
             }
           }
+          // If changing refreshTokenId, the old token loses its device record.
+          if (device.refreshTokenId) {
+            if (deviceInfo.refreshTokenId) {
```

Comment:
What if a device HAD a refreshTokenId and no longer does because a sessionTokenId is passed in instead? Worth worrying about?

Reply:
I think it should wind up with both; we don't have any reason to expect tokens to move from one device to another.

Reply:
Hrm, or do you mean that we're explicitly passing `refreshTokenId == null` in that case?

Reply:
I've added an explicit test that asserts it will wind up with *both* tokens on the device record, does this seem reasonable behaviour?